File: packages/python/src/mickerbook_sdk/redact.py
import re


_MICKERBOOK_KEY = re.compile(r"micker_sk_[A-Za-z0-9_-]+")
_BEARER = re.compile(r"Bearer\s+[A-Za-z0-9._~+/=-]+", re.IGNORECASE)


def redact_secrets(value):
    if isinstance(value, str):
        return _BEARER.sub("Bearer [REDACTED]", _MICKERBOOK_KEY.sub("[REDACTED]", value))
    if isinstance(value, list):
        return [redact_secrets(item) for item in value]
    if isinstance(value, tuple):
        return tuple(redact_secrets(item) for item in value)
    if isinstance(value, dict):
        return {key: redact_secrets(item) for key, item in value.items()}
    return value

File: packages/js/src/redact.js
export function redactSecrets(value) {
  if (typeof value === "string") {
    return value
      .replace(/micker_sk_[A-Za-z0-9_-]+/g, "[REDACTED]")
      .replace(/(Bearer\s+)[A-Za-z0-9._~+/-]+=*/gi, "$1[REDACTED]")
      .replace(/(api[_-]?key["']?\s*[:=]\s*["']?)[^"',\s]+/gi, "$1[REDACTED]")
      .replace(/(token["']?\s*[:=]\s*["']?)[^"',\s]+/gi, "$1[REDACTED]")
      .replace(/(cookie["']?\s*[:=]\s*["']?)[^"',\n]+/gi, "$1[REDACTED]");
  }

  if (Array.isArray(value)) {
    return value.map((item) => redactSecrets(item));
  }

  if (value && typeof value === "object") {
    return Object.fromEntries(
      Object.entries(value).map(([key, item]) => [key, redactSecrets(item)]),
    );
  }

  return value;
}
